kotlin/internal/jvm/plugins.bzl

```diff
@@ -63,6 +78,21 @@ def _kt_jvm_plugin_aspect_impl(target, ctx):
                     generates_api = processor.generates_api,
                 ),
             ]),
+            plugins = depset(),
+        )]
+    elif ctx.rule.kind == "kt_plugin":
```

Comment:
This is a kotlin only rule, so using as aspect to collect is not suggested -- the aspect is to work around the java rules not exporting enough information. Instead, the plugin information should collected via provider.

Reply:
That makes sense. To be clear, I would then create a provider solely for the compiler plugin info? Maybe `KtCompilerPluginInfo`?

Reply:
Yes, that would be good.

Reply:
I added KtCompilerPluginInfo to defs.bzl